gradle-incremental-configuration-cache/src/main/java/com/palantir/gradle/configcache/incremental/DryRunConfigurationCacheEnabledTask.java

```diff
@@ -0,0 +1,210 @@
+
+            %s
+
+            WHY THIS MATTERS:
+              This validation task runs all the tasks marked as configuration cacheable in the allow list,
+              to ensure the allow list contains only cacheable tasks.
+              Regular CI builds may mask configuration cache issues when non-cacheable
+              tasks disable the configuration cache.
+
+            HOW TO FIX (only if you have introduced a task, or upgraded a plugin):
+
+              1. Review the %s for specific issues
+
+              2. Common fixes for configuration cache problems:
+                  • Task.project at execution → Inject services (ProjectLayout, FileSystemOperations)
+                  • External processes → Use ProviderFactory.exec() or GradleExec (https://github.com/palantir/gradle-utils?tab=readme-ov-file#gradleexec)
+                  • Cannot serialize Gradle model types → Don't pass in full object use Property<T> for specific values or inject services
+
+              3. If you upgraded a plugin, verify it supports configuration cache
+
+              📚 Gradle Guide: https://github.com/palantir/gradle-guide/blob/develop/guide/adopting-the-configuration-cache.md
+              📚 Gradle docs: https://docs.gradle.org/current/userguide/configuration_cache.html
+            """,
+                reportsSection,
+                configCacheReportUrl.isPresent() || configCacheReportPath.isPresent()
+                        ? "Gradle configuration cache report"
+                        : "output above");
+
+        // Add validation output for local development only
+        if (validationReportUrl.isEmpty() && outputContent != null && !outputContent.isEmpty()) {
+            mainMessage +=
+                    """
+
+                VALIDATION OUTPUT: to see full output above re-run with --info
+                """;
+            // the outputContent can be very large and annoying to scroll past so only show when running with info
+            getLogger().info(outputContent.orElse(""));
+        }
+
+        return mainMessage;
+    }
+
+    private static String reportsSection(
+            Optional<String> validationReportUrl,
+            Optional<String> configCacheReportUrl,
+            Optional<String> configCacheReportPath) {
+
+        Optional<String> configCacheLocation =
+                configCacheReportUrl.or(() -> configCacheReportPath.map(path -> "file://" + path));
+
+        Optional<String> configCacheReport = configCacheLocation.map(location ->
+                String.format("%s  📊 Config cache report: %s", validationReportUrl.isPresent() ? "\n" : "", location));
+
+        return validationReportUrl
+                .map(url -> String.format("  📋 Full output: %s%s", url, configCacheReport.orElse("")))
+                .orElseGet(() -> configCacheReport.orElse("  To see full output above re-run with --info"));
+    }
+
+    private Optional<String> extractConfigCacheReportPath(Optional<String> output) {
```

Comment:
this is a little bit weird, we should have the method deal with the contents and not worry about the initial optional i.e.

```java
private static Optional<String> extractConfigCacheReportPath(String output) {
  String pattern = "See the complete report at file://";
  // rest of logic
}
```

Then you do: `outputContent.flatMap(extractConfigCacheReportPath)`